internal/pkg/template/overrides.go

```diff
@@ -43,19 +44,37 @@ type CFNResource struct {
 	LogicalID string
 }
 
+// CDKImport is the interface to import a CDK package.
+type CDKImport interface {
+	ImportName() string
+	ImportShortRename() string
+}
+
 type cfnResources []CFNResource
 
-// UniqueTypes returns the list of unique CFN types.
-func (rs cfnResources) UniqueTypes() []CFNType {
-	var uniqueTypes []CFNType
-	seen := make(map[CFNType]struct{})
+// Imports returns a list of CDK imports for a given list of CloudFormation resources.
+func (rs cfnResources) Imports() []CDKImport {
+	// Find a unique CFN type per service.
+	// For example, given "AWS::ECS::Service" and "AWS::ECS::TaskDef" we want to retain only one of them.
+	seen := make(map[string]CFNType)
 	for _, r := range rs {
-		if _, ok := seen[r.Type]; !ok {
-			uniqueTypes = append(uniqueTypes, r.Type)
+		serviceName := strings.Split(strings.ToLower(string(r.Type)), "::")[1]
+		if _, ok := seen[serviceName]; ok {
+			continue
 		}
-		seen[r.Type] = struct{}{}
+		seen[serviceName] = r.Type
+	}
+
+	imports := make([]CDKImport, len(seen))
```

Comment:
How about adding this before for loop and inside for loop just add `imports = append(imports, r.Type)`. This will save extra for logic to create our resultant slice.

Reply:
:rip: sadly if I want to use `append` I need to use `var imports []CDKImport` rather than initializing it with a size ahead of time

https://go.dev/play/p/q6LfsDBih3y